lib/aem/aemupload.js

```diff
@@ -43,7 +44,16 @@ const { FilterFailedAssets } = require("../functions/filterfailedassets");
  */
 async function* generateAEMUploadTransferRecords(options) {
     for (const uploadFile of options.uploadFiles) {
-        const sourceUrl = fileUrl(uploadFile.filePath);
+        let sourceUrl = uploadFile.blob;
+        if (!sourceUrl) {
+            if (!uploadFile.filePath) {
+                throw new IllegalArgumentError(
+                    'Either blob or filePath must be provided in uploadFiles',
+                    JSON.stringify(uploadFile),
```

Comment:
eslint may complain about this if the last comma is not removed:
```suggestion
                    JSON.stringify(uploadFile)
```

Reply:
It didn't complain, but I'll remove the comma anyway.